tests/gtest/avifdimgtest.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+#include "avif/avif.h"
+#include "avifincrtest_helpers.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace avif {
+namespace {
+
+// Used to pass the data folder path to the GoogleTest suites.
+const char* data_path = nullptr;
+
+//------------------------------------------------------------------------------
+
+TEST(IncrementalTest, Dimg) {
+  testutil::AvifRwData avif =
+      testutil::ReadFile(std::string(data_path) + "sofa_grid1x5_420.avif");
+  ASSERT_NE(avif.size, 0u);
+  ImagePtr reference(avifImageCreateEmpty());
+  ASSERT_NE(reference, nullptr);
+  DecoderPtr decoder(avifDecoderCreate());
+  ASSERT_NE(decoder, nullptr);
+  ASSERT_EQ(avifDecoderReadMemory(decoder.get(), reference.get(), avif.data,
+                                  avif.size),
+            AVIF_RESULT_OK);
+
+  // Change the order of the 'dimg' associations.
```

Comment:
this bit twiddling is non-trivial. could we just add a new test file with the swapped dimg ordering?